api/metrics/src/main/java/io/opentelemetry/api/metrics/Meter.java

```diff
@@ -8,174 +8,53 @@
 import javax.annotation.concurrent.ThreadSafe;
 
 /**
- * Meter is a simple, interface that allows users to record measurements (metrics).
+ * Provides instruments used to produce metrics.
  *
- * <p>There are two ways to record measurements:
+ * <p>Instruments are obtained through builders provided by this interface. Each builder has a
+ * default "type" associated with recordings that may be changed.
  *
- * <ul>
- *   <li>Record raw measurements, and defer defining the aggregation and the labels for the exported
- *       Instrument. This should be used in libraries like gRPC to record measurements like
- *       "server_latency" or "received_bytes".
- *   <li>Record pre-defined aggregation data (or already aggregated data). This should be used to
- *       report cpu/memory usage, or simple metrics like "queue_length".
- * </ul>
- *
- * <p>TODO: Update comment.
+ * <p>A Meter is generally associated with an instrumentation library, e.g. "I monitor apache
+ * httpclient".
  */
 @ThreadSafe
 public interface Meter {
```

Comment:
The async instruments are accessible via the `buildWithCallback` methods from the respective builders, rather than from the `Meter`.

Is this allowed according to the spec?

Reply:
I would say so. You still get the builder via the `Meter`. The spec isn't supposed to precisely define the language-specific APIs, but the capabilities that must be provided.

Reply:
This was deemed a Java-specific deviation to abide by the common builder pattern in Java.   I like the simplicity it acheived, but it is something we should discuss and can migrate to having separate builders by-async type instead of a builder per-metric-type and then having the async builder + sync builder coexist.